src/Microsoft.SqlTools.ServiceLayer/Agent/Common/AgentUtilities.cs

```diff
@@ -82,9 +78,9 @@ public static AgentNotebookInfo ConvertToAgentNotebookInfo(JobProperties job)
                 Category = job.Category,
                 CategoryId = job.CategoryID,
                 CategoryType = job.CategoryType,
-                LastRun = job.LastRun != null ? job.LastRun.ToString() : string.Empty,
-                NextRun = job.NextRun != null ? job.NextRun.ToString() : string.Empty,
-                JobId = job.JobID != null ? job.JobID.ToString() : null,
+                LastRun = job.LastRun.ToString(),
```

Comment:
Hrm, I'm a bit hesitant about these null changes. It seems odd that these checks would have been added here without a reason - and for jobs it's possible for the "last run" date to be null (never ran).

Could you verify for all these that they're populated in such a way that they'd never be null?

Reply:
I believe it's a new C# Feature 8.0 feature. Essentially objects have to be marked as nullable in order to ever be null.

https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references

Reply:
Yeah, but is that actually guaranteed? We should make sure to check how these values are loaded in - it may be that we're just blindly casting them to non-nullable types somewhere or something else like that.

Reply:
I dug into the code more to see where these values get set.. here is the code for the backing field. Based on this logic, if the value was null it would throw a null reference exception at this level (higher in the stack).

> this.jobId                   = Guid.Parse(row["JobID"].ToString());
    if (row["LastRunDate"] != DBNull.Value)
    {
        this.lastRun = Convert.ToDateTime(row["LastRunDate"], CultureInfo.InvariantCulture);
    }

    if (row["NextRunDate"] != DBNull.Value)
    {
        this.nextRun = Convert.ToDateTime(row["NextRunDate"], CultureInfo.InvariantCulture);
    }